### PR TITLE
Style and split

### PR DESCRIPTION
**Pull request by tdresser**

### CODE REVIEW

File: cpp/action.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                        {ActionType::TouchStart, "touchstart"},
+                        {ActionType::TouchMove, "touchmove"},
+                    });
+
+class Action {
+ public:
+  Action(ActionType type, Point point) : _type(type), _point(point) {}
+  Action() = default;
+  ;
+  // TODO(tdresser) - make this virtual, and use polymorphism?
```
Comment (tdresser) [inline cpp/action.h]: Having thought more, this is probably a bad idea.

File: .clang-tidy [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  -fuchsia-statically-constructed-objects,
+  -*struct-pack-align,
+  -cert-err58-cpp,
+  -cppcoreguidelines-avoid-non-const-global-variables,
+  -readability-convert-member-functions-to-static,
+  -cppcoreguidelines-macro-usage,
+  -cppcoreguidelines-pro-type-reinterpret-cast,
+  -performance-no-int-to-ptr,
+
+# Don't forget the trailing comma above.
```
Comment (tdresser) [inline .clang-tidy:25]: TODO - double check this...

File: cpp/emscripten_util.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+EMSCRIPTEN_BINDINGS(Bindings) {
+  emscripten::function("getExceptionMessage", &getExceptionMessage);
+};
+
+#define TO_JSON(CLASSNAME,                                               \
+                INSTANCENAME) /* NOLINT(readability-identifier-naming)*/ \
+  void to_json(json& j, const CLASSNAME&(INSTANCENAME)) {                \
+    (INSTANCENAME).toJson(j);                                            \
+  };
```
Comment (tdresser) [inline cpp/emscripten_util.h]: These defines are kinda gross, but it centralizes the NOLINT, which I like.

File: cpp/action.h (hunk 1 above)
Comment (ojanvafai) [reply cpp/action.h]: I dno't feel strongly, but I think I agree.

File: cpp/emscripten_util.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  return std::string(reinterpret_cast<std::exception*>(exceptionPtr)->what());
+}
+
+EMSCRIPTEN_BINDINGS(Bindings) {
+  emscripten::function("getExceptionMessage", &getExceptionMessage);
+};
+
+#define TO_JSON(CLASSNAME,                                               \
+                INSTANCENAME) /* NOLINT(readability-identifier-naming)*/ \
+  void to_json(json& j, const CLASSNAME&(INSTANCENAME)) {                \
```
Comment (ojanvafai) [inline cpp/emscripten_util.h]: What calls this? Is it inside of nlohmann::json?

**Review by ojanvafai — APPROVED** [review]
lgtm!

File: cpp/action.h [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+                        {ActionType::Unknown, "unknown"},
+                        {ActionType::TouchStart, "touchstart"},
+                        {ActionType::TouchMove, "touchmove"},
+                    });
+
+class Action {
+ public:
+  Action(ActionType type, Point point) : _type(type), _point(point) {}
+  Action() = default;
+  ;
```
Comment (tdresser) [inline cpp/action.h]: Remove ;.

File: cpp/emscripten_util.h (hunk 4 above)
Comment (tdresser) [reply cpp/emscripten_util.h]: Yup. for example, this is why we in State, we can just do:
  void toJson(json& j) const { j = json{{"transform", _transform}}; }

That implicitly calls to_json on the point object to convert _transform to json.

**Pull request merged**